app/helpers/common_parameters_helper.rb

```diff
@@ -70,9 +70,14 @@ def lookup_key_field(id, value, options)
 
   def authorized_resource_parameters(resource, type)
     parameters_by_type = resource.send(type)
-    parameter_ids_to_view = parameters_by_type.authorized(:view_params).map(&:id)
-    resource_parameters = parameters_by_type.select { |p| parameter_ids_to_view.include?(p.id) }
+    @authorizer_for_params = Authorizer.new(User.current, :collection => parameters_by_type)
+    resource_parameters = @authorizer_for_params.find_collection(
+      parameters_by_type, :permission => :view_params)
```

Comment:
I think that `parameters_by_type` should be replaced by parameter type here. `find_collection` needs a resource type as the first parameter, not a collection.